app/components/embedded_bindings/menu_binding/menu_binding.tsx

```diff
@@ -83,6 +82,8 @@ export default class MenuBinding extends PureComponent<Props, State> {
         );
 
         const res = await actions.doAppCall(call, AppCallTypes.SUBMIT, this.context.intl);
+
+        const ephemeral = (message: string) => this.props.actions.sendEphemeralPost(message, this.props.post.channel_id, this.props.post.root_id, res.data?.app_metadata?.bot_user_id);
```

Comment:
Are we ok to wait until response comes back?

Reply:
> Are we checking that message is not empty before adding the ephemeral post?

@enahum Yes there is a check below for the message. This line is actually just making a shorthand function to call to include the relevant arguments to `sendEphemeralPost`.

> Does it make sense to add the ephemeral post here at the client than throw the WS?

Yes, all of the call response ephemeral posts are made client-side. This is because depending on the context, the response may be displayed differently, including what UI elements triggered the call and what platform you are using.

> Are we ok to wait until response comes back?

I'm not sure what you mean. We don't have anything to post until the response comes back.